src/crontab.c

```diff
@@ -479,16 +479,78 @@ static void delete_cmd(void) {
 		fprintf(stderr, "path too long\n");
 		exit(ERROR_EXIT);
 	}
+	
+	if(backup_crontab(n) == -1){
+		fprintf(stderr, "no crontab for %s\n", User);
+		exit(ERROR_EXIT);
+	}
+
 	if (unlink(n) != 0) {
-		if (errno == ENOENT)
-			fprintf(stderr, "no crontab for %s\n", User);
-		else
-			perror(n);
+		perror(n);
 		exit(ERROR_EXIT);
 	}
 	poke_daemon();
 }
 
+/* returns	0	on success
+ *		-1	on non existent crontab file
+ */
+static int backup_crontab(char crontab_filepath[]) {
+	const char *env_value;
+	char backup_filename[MAX_FNAME];
+	
+	if(env_value = getenv("XDG_CACHE_HOME")){
+		if (!glue_strings(backup_filename, sizeof backup_filename, env_value,
+			"crontab.bak", '/')) {
+				fprintf(stderr, "path too long\n");
+				exit(ERROR_EXIT);
+		}
+	}else if(env_value = getenv("HOME")){
+		if (!glue_strings(backup_filename, sizeof backup_filename, env_value,
+			".cache/crontab.bak", '/')) {
+				fprintf(stderr, "path too long\n");
+				exit(ERROR_EXIT);
+		}
+	}else{
+		fprintf(stderr, "Could not find environment variable XDG_CACHE_HOME or HOME to save the backup\n");
+		exit(ERROR_EXIT);
+	}
+
+	FILE *crontab_file;
+	FILE *backup_file;
+
+	if (!(crontab_file = fopen(crontab_filepath, "r"))) {
+		if (errno != ENOENT) {
+			perror(crontab_filepath);
+			exit(ERROR_EXIT);
+		}
+		return -1;
+	}
+	
+	if (swap_uids() == -1) {
+		perror("swapping uids");
+		exit(ERROR_EXIT);
+	}
+	if (!(backup_file = fopen(backup_filename, "w+"))) {
+		perror(backup_filename);
+		exit(ERROR_EXIT);
```

Comment:
Inability to write the backup file should just print a warning message (which should be more explanatory) and should not cause an error exit.

Reply:
I'm really unsure about what I did, you should take take a look : [8589910](https://github.com/cronie-crond/cronie/pull/128/commits/8589910600c9f9dbde505ce690aff14d95ddabce)